graphjet-core/src/main/java/com/twitter/graphjet/algorithms/TweetAuthorFilter.java

```diff
@@ -65,6 +68,6 @@ public void resetFilter(RecommendationRequest request) {}
    */
   @Override
   public boolean filterResult(long resultNode, SmallArrayBasedLongToDoubleMap[] socialProofs) {
-    return !authoredTweets.contains(resultNode);
+    return hasAuthors && !authoredTweets.contains(resultNode);
```

Comment:
The behavior here is actually a bit different. When initializing this Filter, we check for all the tweets authored by a list of authors, and toss them into a list called authoredTweets.
If we don't specify a list of authors in the beginning, it means we don't require tweets to be authored from anyone. This will always result in an empty authoredTweets. This is different from specifying a list of authors, and not find any tweets authored from them, which also results in an empty authoredTweets. Let me add a comment to clarify this behavior. Sorry about the confusion